R/components.R

```diff
@@ -0,0 +1,19 @@
+#' Extract components
+#'
+#' Decomposition methods and some modelling techniques can be used to decompose
```

Comment:
To make this a little more general, maybe 

> `components` can be used to extract relavant elements of an object. For example, decomposition methods and some modelling techniques can be used to decompose a dataset into components of interest. This function can used to extract these components in a tidy data format.

I could see myself using this for applications that are not exactly the same as the listed intent.

Reply:
Making it more general is fine.